ext/ddtrace_profiling_native_extension/stack_recorder.c

```diff
@@ -551,14 +554,34 @@ static bool add_heap_sample_to_active_profile_without_gvl(heap_recorder_iteratio
   int64_t metric_values[ALL_VALUE_TYPES_COUNT] = {0};
   uint8_t *position_for = context->state->position_for;
 
-  metric_values[position_for[HEAP_SAMPLES_VALUE_ID]] = iteration_data.object_data.weight;
+  metric_values[position_for[HEAP_SAMPLES_VALUE_ID]] = iteration_data.object_data->weight;
+
+  const live_object_data *object_data = iteration_data.object_data;
+  ddog_prof_Label labels[2];
+  size_t num_labels = 2;
```

Comment:
Minor:

```suggestion
  size_t num_labels = 2;
  ddog_prof_Label labels[num_labels];
```

Reply:
Needed a slight re-work to deal with optional class names.